src/Symfony/Component/VarDumper/Server/DumpServer.php

```diff
@@ -49,33 +55,18 @@ public function start(): void
         }
     }
 
-    public function listen(callable $callback): void
+    public function listen(callable $callback, ?StreamableInputInterface $streamInput): void
     {
-        if (null === $this->socket) {
+        if ($streamInput instanceof StreamableInputInterface && $stream = $streamInput->getStream()) {
+            $this->inputStream = $stream;
+        } elseif (null === $this->socket) {
             $this->start();
         }
 
         foreach ($this->getMessages() as $clientId => $message) {
             $this->logger?->info('Received a payload from client {clientId}', ['clientId' => $clientId]);
 
-            $payload = @unserialize(base64_decode($message), ['allowed_classes' => [Data::class, Stub::class]]);
-
-            // Impossible to decode the message, give up.
-            if (false === $payload) {
-                $this->logger?->warning('Unable to decode a message from {clientId} client.', ['clientId' => $clientId]);
-
-                continue;
-            }
-
-            if (!\is_array($payload) || \count($payload) < 2 || !$payload[0] instanceof Data || !\is_array($payload[1])) {
-                $this->logger?->warning('Invalid payload from {clientId} client. Expected an array of two elements (Data $data, array $context)', ['clientId' => $clientId]);
-
-                continue;
-            }
-
-            [$data, $context] = $payload;
-
-            $callback($data, $context, $clientId);
+            $callback($clientId, $message);
```

Comment:
that's a BC break as this changes the contract of the callback.
Also I'm not sure this makes sense it terms of responsibility.

Reply:
I did this following [your comment](https://github.com/symfony/symfony/pull/53518#issuecomment-1914261636) in the PR #53518.  
Do you have an idea ?  
To help you, here [a try](https://github.com/louismariegaborit/symfony/commit/cbbb48fa8c9f94d78c0951140db58d034e06a8f7) of usage on the monolog bridge to see possibilities.